server/src/main/java/org/opensearch/gateway/remote/RemoteClusterStateAttributesManager.java

```diff
@@ -0,0 +1,195 @@
+import org.opensearch.cluster.block.ClusterBlocks;
+import org.opensearch.cluster.node.DiscoveryNodes;
+import org.opensearch.common.CheckedRunnable;
+import org.opensearch.common.remote.AbstractRemoteWritableBlobEntity;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.core.common.io.stream.NamedWriteableRegistry;
+import org.opensearch.core.compress.Compressor;
+import org.opensearch.core.xcontent.NamedXContentRegistry;
+import org.opensearch.core.xcontent.ToXContent;
+import org.opensearch.gateway.remote.RemoteClusterStateUtils.RemoteStateTransferException;
+import org.opensearch.gateway.remote.model.RemoteClusterBlocks;
+import org.opensearch.gateway.remote.model.RemoteClusterStateBlobStore;
+import org.opensearch.gateway.remote.model.RemoteClusterStateCustoms;
+import org.opensearch.gateway.remote.model.RemoteDiscoveryNodes;
+import org.opensearch.gateway.remote.model.RemoteReadResult;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import static org.opensearch.gateway.remote.RemoteClusterStateUtils.CUSTOM_DELIMITER;
+import static org.opensearch.gateway.remote.model.RemoteClusterStateCustoms.CLUSTER_STATE_CUSTOM;
+
+/**
+ * A Manager which provides APIs to upload and download attributes of ClusterState to the {@link RemoteClusterStateBlobStore}
+ *
+ * @opensearch.internal
+ */
+public class RemoteClusterStateAttributesManager {
+    public static final String CLUSTER_STATE_ATTRIBUTE = "cluster_state_attribute";
+    public static final String DISCOVERY_NODES = "nodes";
+    public static final String CLUSTER_BLOCKS = "blocks";
+    public static final int CLUSTER_STATE_ATTRIBUTES_CURRENT_CODEC_VERSION = 1;
+    private final RemoteClusterStateBlobStore<ClusterBlocks, RemoteClusterBlocks> clusterBlocksBlobStore;
+    private final RemoteClusterStateBlobStore<DiscoveryNodes, RemoteDiscoveryNodes> discoveryNodesBlobStore;
+    private final RemoteClusterStateBlobStore<Custom, RemoteClusterStateCustoms> customsBlobStore;
+    private final Compressor compressor;
+    private final NamedXContentRegistry namedXContentRegistry;
+    private final NamedWriteableRegistry namedWriteableRegistry;
+
+    RemoteClusterStateAttributesManager(
+        RemoteClusterStateBlobStore<ClusterBlocks, RemoteClusterBlocks> clusterBlocksBlobStore,
+        RemoteClusterStateBlobStore<DiscoveryNodes, RemoteDiscoveryNodes> discoveryNodesBlobStore,
+        RemoteClusterStateBlobStore<Custom, RemoteClusterStateCustoms> customsBlobStore,
+        Compressor compressor,
+        NamedXContentRegistry namedXContentRegistry,
+        NamedWriteableRegistry namedWriteableRegistry
+    ) {
+        this.clusterBlocksBlobStore = clusterBlocksBlobStore;
+        this.discoveryNodesBlobStore = discoveryNodesBlobStore;
+        this.customsBlobStore = customsBlobStore;
+        this.compressor = compressor;
+        this.namedXContentRegistry = namedXContentRegistry;
```

Comment:
xcontentRegistry is not needed now

Reply:
We need to create objects of `AbstractRemoteWritableBlobEntity` where passing the namedXContentRegistry is required. Should I modify the constructor of `AbstractRemoteWritableBlobEntity` to enable us to create a object with namedWritableRegistry rather than XContentRegistry.

Reply:
We can create a new constructor. But for now even passing namedXContentRegistry as null should be fine.